include/na/Architecture.hpp

```diff
@@ -0,0 +1,290 @@
+   * - T2
+   * - effective decoherence time
+   */
+  struct DecoherenceTimes {
+    Value t1                    = 0;
+    Value t2                    = 0;
+    Value tEff                  = 0;
+    DecoherenceTimes()          = default;
+    virtual ~DecoherenceTimes() = default;
+    DecoherenceTimes(const Value t1time, const Value t2time)
+        : t1(t1time), t2(t2time), tEff(t1 * t2 / (t1 + t2)) {}
+    explicit operator double() const { return tEff; }
+  };
+  /**
+   * @brief Strcut to store the properties of an operation.
+   * @details Times are in µs, fidelities are in [0,1].
+   */
+  struct OperationProperties {
+    Scope                    scope; // local or global
+    std::unordered_set<Zone> zones; // the zones where the gate can be applied
+    Value time;     // the time the gate takes to be applied in µs
+    Value fidelity; // the fidelity of the gate
+  };
+  /**
+   * @brief Struct to store the properties of a Shuttling operation (i.e. of the
+   * AOD).
+   * @details Times are in µs, fidelities are in [0,1], and velocities are in
+   * µm/µs.
+   */
+  struct ShuttlingProperties {
+    Number rows          = 0; // maximum number of rows in one AOD
+    Number cols          = 0; // maximum number of columns in one AOD
+    Number minX          = 0; // minimum x position of the AOD
+    Number maxX          = 0; // maximum x position of the AOD
+    Number minY          = 0; // minimum y position of the AOD
+    Number maxY          = 0; // maximum y position of the AOD
+    Value  speed         = 0; // speed of the AOD in µm/µs
+    Value  fidelity      = 1; // fidelity during the shuttling
+    Value  loadTime      = 0; // time to activate the AOD in µs
+    Value  loadFidelity  = 1; // fidelity of the load
+    Value  storeTime     = 0; // time to deactivate the AOD in µs
+    Value  storeFidelity = 1; // fidelity of the store
+  };
+  struct ZoneProperties {
+    std::string name;         // the name of the zone
+    Number      minX     = 0; // minimum x dimension
+    Number      maxX     = 0; // maximum x dimension
+    Number      minY     = 0; // minimum y dimension
+    Number      maxY     = 0; // maximum y dimension
+    Value       fidelity = 1; // fidelity during idling
+  };
+
+protected:
+  std::string name; // the name of the architecture
+  std::vector<ZoneProperties>
+      zones; // a mapping from zones (int) to their name from the config
+  std::vector<Point> sites; // a vector of sites
+  std::unordered_map<FullOpType, OperationProperties>
+      gateSet; // all possible operations by their type, i.e. gate set
+  DecoherenceTimes decoherenceTimes;          // the decoherence characteristic
+  std::vector<ShuttlingProperties> shuttling; // all properties regarding AODs
+  Index minAtomDistance = 0; // minimal distance that must be kept between atoms
+  Index interactionRadius = 0; // the Rydberg radius
+  Index noInteractionRadius =
+      0; // sufficient radius to avoid Rydberg interaction
```

Comment:
I changed it to Value but figured that I have not done it consistently yet. In the meantime I am thinking whether this is actually a good idea to convert it to a floating point number. Assume that you want to find the nearest site left of the position (1.99999999, 0). You actually want to find the site (2,0) because the 1.999999999 only came up due to a rounding error.

When I wrote this code initially, I decided that µm is small enough to define locations. What do you say with repsect to my concern?